src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,39 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileReader;
+import java.io.IOException;
+import java.nio.file.Files;
+
 public class WorkWithFile {
-    public void getStatistic(String fromFileName, String toFileName) {
 
+    public void getStatistic(String fromFileName, String toFileName) {
+        File sourceFile = new File(fromFileName);
+        File destinationfile = new File(toFileName);
+        try (BufferedReader reader = new BufferedReader(new FileReader(sourceFile))) {
+            String readerString = reader.readLine();
+            int supplyCount = 0;
+            int buyCount = 0;
+            int operatedAmount;
+            String[] splittedLine;
+            while (readerString != null) {
+                splittedLine = readerString.split(",");
+                operatedAmount = Integer.parseInt(splittedLine[1]);
+                if (splittedLine[0].equals("supply")) {
+                    supplyCount += operatedAmount;
+                } else {
+                    buyCount += operatedAmount;
+                }
+                readerString = reader.readLine();
+            }
+            String resultData = "supply," + supplyCount + System.lineSeparator()
+                    + "buy," + buyCount + System.lineSeparator()
+                    + "result," + (supplyCount - buyCount);
```

Comment:
[Use StringBuilder instead of String to concat data in loops.](https://mate-academy.github.io/jv-program-common-mistakes/java-core/builder-file/work-with-file#dont-ignore-exceptions)

Reply:
Ok, but it's not a loop.